react/react-training/src/app/screens/Game/index.js

```diff
@@ -7,47 +10,34 @@ import HistoryItem from './components/HistoryItem/';
 import styles from './styles.scss';
 
 class Game extends Component {
-  state = {
-    history: [
-      {
-        squares: Array(9).fill(null)
-      }
-    ],
-    stepNumber: 0,
-    xIsNext: true
-  };
-
   handleClick = i => {
-    const history = this.state.history.slice(0, this.state.stepNumber + 1);
-    const current = history[history.length - 1];
+    const { history, stepNumber, xIsNext, makeMove } = this.props;
+    const newHistory = history.slice(0, stepNumber + 1);
+    const current = newHistory[newHistory.length - 1];
     const squares = current.squares.slice();
 
     if (calculateWinner(squares) || squares[i]) {
       return;
     }
-    squares[i] = this.state.xIsNext ? 'X' : 'O';
-    this.setState({
-      history: history.concat([{ squares }]),
-      stepNumber: history.length,
-      xIsNext: !this.state.xIsNext
-    });
+
+    squares[i] = xIsNext ? 'X' : 'O';
```

Comment:
I think that you can calculate this value in `makeMove` action

Reply:
Ignoro este cambio.